### PR TITLE
add project learning objectives doc

### PR DESCRIPTION
**Pull request by richardaspinall**
What?
Create a doc that has our project learning objectives fleshed out

Why?
So we can anchor on specific learnings when we are working the project

ClickUp task:
https://app.clickup.com/t/860qc32xa

### CODE REVIEW

File: docs/project_learning_objectives.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# [Draft] Project Learning Objectives
+
+We have come up with a list of learning objectives for this project, but it's a dynamic ever growing list. If we have something we are wanting to learn currently, then we can add that as an objective and try and come with some work or feature that anchors on the learning objective. For now, we have added them as labels that can be added to tasks and PRs:
+
+---
+
+## MVP / First phase
```
Comment (richardaspinall) [inline docs/project_learning_objectives.md]: What I'm thinking is that we change this to "MVP – make it work" and then we remove a couple of the learning objectives from this phase such as "Object Oriented Learning", "Design & Architecture" and "Scalability". 

Then we add those learning objectives to the next phase after we have something working. What do you reckon?

File: docs/project_learning_objectives.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+| Project Learning Objective    | Description                                                                       |
+| ----------------------------- | --------------------------------------------------------------------------------- |
+| Git and Github                | How to use Git and Github and some best practices                                 |
+| Code Review                   | How to review code of peers as quality assurance, learn and help each other learn |
+| Collaboration                 | How to work together on a piece of work or task including pair programming        |
+| Team work                     | How to divide and conquer tasks combining individual efforts to achieve goals     |
+| Test Driven Development (TDD) | How to do test driven development                                                 |
+| Feature Requirement           | How to create, scope and discover requirements for features                       |
+| Backend                       | How to code for the backend                                                       |
+| Object Oriented Learning      | How to write in an Object Oriented Programming way                                |
```
Comment (richardaspinall) [inline docs/project_learning_objectives.md]:
```suggestion
```
As per: https://github.com/richardaspinall/poker/pull/1/files#r1150363680

File: docs/project_learning_objectives.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+| Design & Architecture         | How to design and architect a system                                              |
+| TypeScript                    | How to write in Typescript which is strongly typed JavaScript                     |
+| Unit Testing                  | How to create unit tests which is testing the smallest testable parts of a system |
+| Scalability                   | How to scale and create code that scales                                          |
```
Comment (richardaspinall) [inline docs/project_learning_objectives.md]:
```suggestion
```
As per: https://github.com/richardaspinall/poker/pull/1/files#r1150363680

File: docs/project_learning_objectives.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| ----------------------------- | --------------------------------------------------------------------------------- |
+| Git and Github                | How to use Git and Github and some best practices                                 |
+| Code Review                   | How to review code of peers as quality assurance, learn and help each other learn |
+| Collaboration                 | How to work together on a piece of work or task including pair programming        |
+| Team work                     | How to divide and conquer tasks combining individual efforts to achieve goals     |
+| Test Driven Development (TDD) | How to do test driven development                                                 |
+| Feature Requirement           | How to create, scope and discover requirements for features                       |
+| Backend                       | How to code for the backend                                                       |
+| Object Oriented Learning      | How to write in an Object Oriented Programming way                                |
+| Design & Architecture         | How to design and architect a system                                              |
```
Comment (richardaspinall) [inline docs/project_learning_objectives.md]:
```suggestion 
```
As per https://github.com/richardaspinall/poker/pull/1/files#r1150363680

File: docs/project_learning_objectives.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+|                               |                                                                                   |
+
+---
+
+## Later
+
+| Project Learning Objective | Description                                                      |
+| -------------------------- | ---------------------------------------------------------------- |
+| Docker                     | How to create docker images and run a consistent dev environment |
+|                            |                                                                  |
```
Comment (richardaspinall) [inline docs/project_learning_objectives.md:33]:
```suggestion
| Object Oriented Learning                    |     How to write in an Object Oriented Programming way                                                             |
| Design & Architecture | How to design and architect a system  |
| Scalability | How to scale and create code that scales |
```
As per: https://github.com/richardaspinall/poker/pull/1/files#r1150363680

File: docs/project_learning_objectives.md (hunk 1 above)
Comment (richardaspinall) [inline docs/project_learning_objectives.md]:
```suggestion
## MVP – make it work
```

File: docs/project_learning_objectives.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# [Draft] Project Learning Objectives
+
+We have come up with a list of learning objectives for this project, but it's a dynamic ever growing list. If we have something we are wanting to learn currently, then we can add that as an objective and try and come with some work or feature that anchors on the learning objective. For now, we have added them as labels that can be added to tasks and PRs:
```
Comment (tommy-trombone) [inline docs/project_learning_objectives.md:3]:
```suggestion
We have come up a list of learning objectives for this project, but it's a dynamic ever growing list. If we have something we are wanting to learn currently, then we can add that as an objective and try and come with some work or feature that anchors on the learning objective. For now, we have added them as labels that can be added to tasks and PRs:
```

File: docs/project_learning_objectives.md (hunk 1 above)
Comment (tommy-trombone) [inline docs/project_learning_objectives.md]:
```suggestion
## Phase One - Make it work
```
Let's go with a phrase instead of "MVP" as discussed

Comment (tommy-trombone) [reply docs/project_learning_objectives.md]: Yeah that makes it easy to compartmentalise the objectives and have a more concise set for each phase.

Comment (tommy-trombone) [reply docs/project_learning_objectives.md:7]: Yeah that makes it easy to compartmentalise the objectives and have a more concise set for each phase.

File: docs/project_learning_objectives.md (hunk 2 above)
Comment (tommy-trombone) [reply docs/project_learning_objectives.md]: 👍

File: docs/project_learning_objectives.md (hunk 4 above)
Comment (tommy-trombone) [reply docs/project_learning_objectives.md]: 👍

File: docs/project_learning_objectives.md (hunk 3 above)
Comment (tommy-trombone) [reply docs/project_learning_objectives.md]: 👍

File: docs/project_learning_objectives.md (hunk 5 above)
Comment (tommy-trombone) [reply docs/project_learning_objectives.md:33]: 👍

**Review by tommy-trombone — APPROVED** [review]
Made a couple of comments but it's good to go either way.

**Pull request merged**